SimG4Core/Notification/interface/TrackInformation.h

```diff
@@ -52,8 +56,24 @@ class TrackInformation : public G4VUserTrackInformation {
   void setCaloSurfaceParticlePID(int id) { caloSurfaceParticlePID_ = id; }
   double caloSurfaceParticleP() const { return caloSurfaceParticleP_; }
   void setCaloSurfaceParticleP(double p) { caloSurfaceParticleP_ = p; }
-  int getIDfineCalo() const { return ((idFineCalo_ > 0) ? idFineCalo_ : idOnCaloSurface_); }
-  void setIDfineCalo(int id) { idFineCalo_ = id; }
+
+  // Boundary crossing variables
+  void setCrossedBoundary(const G4Track *track) {
+    crossedBoundary_ = true;
+    idAtBoundary_ = track->GetTrackID();
+    positionAtBoundary_ = math::XYZTLorentzVectorF(track->GetPosition().x() / CLHEP::cm,
+                                                   track->GetPosition().y() / CLHEP::cm,
+                                                   track->GetPosition().z() / CLHEP::cm,
+                                                   track->GetGlobalTime());
+    momentumAtBoundary_ = math::XYZTLorentzVectorF(track->GetMomentum().x() / CLHEP::GeV,
+                                                   track->GetMomentum().y() / CLHEP::GeV,
+                                                   track->GetMomentum().z() / CLHEP::GeV,
+                                                   track->GetKineticEnergy() / CLHEP::GeV);
+  }
+  bool crossedBoundary() const { return crossedBoundary_; }
+  const math::XYZTLorentzVectorF getPositionAtBoundary() const { return positionAtBoundary_; }
+  const math::XYZTLorentzVectorF getMomentumAtBoundary() const { return momentumAtBoundary_; }
```

Comment:
This would be a little bit better by reference (to follow the general pattern)
```suggestion
  const math::XYZTLorentzVectorF& getPositionAtBoundary() const { return positionAtBoundary_; }
  const math::XYZTLorentzVectorF& getMomentumAtBoundary() const { return momentumAtBoundary_; }
```
although the practical impact is likely none (compiler can optimize).